### PR TITLE
Data source plugin support xugu database

### PR DESCRIPTION
**Pull request by mrliufox**
## Purpose of the pull request

Data source plugin support xugu database

## Brief change log

Data source plugin support xugu database

## Verify this pull request

This change added tests and can be verified as follows:

- *Added XuguDataSourceChannelFactoryTest to verify the change.*
- *Added XuguDataSourceChannelTest to verify the change.*

### CODE REVIEW

**Comment by mrliufox** [comment]
_❌ (Please check if there are PRs that already have a ready-to-merge label and can be merged, if exists please merge them first.)_

**But I have no merging branches**

**Comment by mrliufox** [comment]
_❌ (Please check if there are PRs that already have a ready-to-merge label and can be merged, if exists please merge them first.)_

 **I have no merging branches**

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-xugu/.gitignore
Comment (EricGao888) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-xugu/.gitignore]: please remove this file

**Comment by mrliufox** [comment]
_❌ (Please check if there are PRs that already have a ready-to-merge label and can be merged, if exists please merge them first.)_

**I have no merging branches**

**Review by songjianet — DISMISSED** [review]

**Review by SbloodyS — CHANGES_REQUESTED** [review]
Related PR: https://github.com/apache/dolphinscheduler/pull/14929

**Comment by mrliufox** [comment]
_Error:  One or more dependencies were identified with vulnerabilities that have a CVSS score greater than or equal to '7.0': _

Please guide me on how to deal with this problem, thank you very much.

**Comment by mrliufox** [comment]
what should i do next?

**Comment by mrliufox** [comment]
_❌ (Please check if there are PRs that already have a ready-to-merge label and can be merged, if exists please merge them first.)_

**I don't have a branch merging.**

**Comment by SbloodyS** [comment]
I haven't found any mail in dev mail list. Please send an email to [dev@dolphinscheduler.apache.org](mailto:dev@dolphinscheduler.apache.org) and explain the significance of supporting xugu database in apache dolphinscheduler. @mrliufox

File: dolphinscheduler-ui/pnpm-lock.yaml
Comment (Gallardot) [inline dolphinscheduler-ui/pnpm-lock.yaml]: Why delete this file?

**Comment by mrliufox** [comment]
> I haven't found any mail in dev mail list. Please send an email to [dev@dolphinscheduler.apache.org](mailto:dev@dolphinscheduler.apache.org) and explain the significance of supporting xugu database in apache dolphinscheduler. @mrliufox

I sent it on February 19, 2024 and it shows that the delivery was successful. Please confirm whether it was intercepted in the trash can.
Snipaste_2024-02-23_14-18-39

File: dolphinscheduler-ui/pnpm-lock.yaml
Comment (mrliufox) [reply dolphinscheduler-ui/pnpm-lock.yaml]: I have restored this file.

**Comment by SbloodyS** [comment]
Maybe you did not subscribe dev mail list. You should subscribe it first. @mrliufox

**Comment by mrliufox** [comment]
> Maybe you did not subscribe dev mail list. You should subscribe it first. @mrliufox

how to subscribe dev mail list? Please guide, thank you very much.

**Comment by SbloodyS** [comment]
You can take a look at https://github.com/apache/dolphinscheduler/issues/8847. @mrliufox

**Comment by mrliufox** [comment]
> You can take a look at #8847. @mrliufox

Thank you very much, I have subscribed to the dev email list and sented an email to dev@dolphinscheduler.apache.org to explain the importance of dolphinscheduler for xugu database support

**Comment by mrliufox** [comment]
**what should i do next?**

**Comment by mrliufox** [comment]
**help please ! what should i do next**

**Comment by mrliufox** [comment]
what should i do next ?

**Comment by SbloodyS** [comment]
Currently, there are no maintainer is willing to accept the maintenance of this data source. So this PR will be temporarily put on hold until a maintainer is willing to continue maintaining it. @mrliufox

**Review by fuchanghai — CHANGES_REQUESTED** [review]
image
You may need to add some logic in the datax plugin

File: dolphinscheduler-spi/src/main/java/org/apache/dolphinscheduler/spi/enums/DbType.java
Comment (fuchanghai) [inline dolphinscheduler-spi/src/main/java/org/apache/dolphinscheduler/spi/enums/DbType.java]: maybe we can use  ``` XUGU(27, "xugu") ```  instead of ```  XUGU(27, "xugu"),```.

**Review by fuchanghai — APPROVED** [review]
basically LGTM  cc @SbloodyS @Gallardot @EricGao888 PTAL

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-xugu/src/main/java/org/apache/dolphinscheduler/plugin/datasource/xugu/param/XuguDataSourceProcessor.java
Comment (wangxj3) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-xugu/src/main/java/org/apache/dolphinscheduler/plugin/datasource/xugu/param/XuguDataSourceProcessor.java:1]: Have you verified the verification of input parameters and output parameters by the sql node?

**Review by wangxj3 — APPROVED** [review]

**Review by zhongjiajie — APPROVED** [review]
LGTM, but it seems @SbloodyS have other oppoin

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-xugu/src/main/java/org/apache/dolphinscheduler/plugin/datasource/xugu/param/XuguDataSourceProcessor.java
Comment (mrliufox) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-xugu/src/main/java/org/apache/dolphinscheduler/plugin/datasource/xugu/param/XuguDataSourceProcessor.java:1]: yes ，I refer to other data sources such as mysql and dameng for parameter verification

**Pull request closed without merging**
